/*
 * This file was automatically generated by EvoSuite
 * Sat Nov 20 21:28:24 GMT 2021
 */

package com.facebook.presto.hive.metastore;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.facebook.presto.hive.metastore.MetastoreContext;
import com.facebook.presto.spi.security.ConnectorIdentity;
import java.util.Optional;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class MetastoreContext_ESTest extends MetastoreContext_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      MetastoreContext metastoreContext0 = null;
      try {
        metastoreContext0 = new MetastoreContext("#j`_H|>V'9(ln2^w", "", (Optional<String>) null, (Optional<String>) null, true, (Optional<String>) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // clientInfo is null
         //
         verifyException("java.util.Objects", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      ConnectorIdentity connectorIdentity0 = mock(ConnectorIdentity.class, new ViolatedAssumptionAnswer());
      doReturn("").when(connectorIdentity0).getUser();
      MetastoreContext metastoreContext0 = null;
      try {
        metastoreContext0 = new MetastoreContext(connectorIdentity0, "", (Optional<String>) null, (Optional<String>) null, (Optional<String>) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // clientInfo is null
         //
         verifyException("java.util.Objects", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      MetastoreContext metastoreContext0 = null;
      try {
        metastoreContext0 = new MetastoreContext("7cn=.Hl", "7cn=.Hl", (Optional<String>) null, (Optional<String>) null, (Optional<String>) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // clientInfo is null
         //
         verifyException("java.util.Objects", e);
      }
  }
}
